hiero-observability/hiero-metrics-core/docs/metrics_overview.md

```diff
@@ -0,0 +1,88 @@
+## Metrics Overview
+
+Metrics are one of the three pillars of observability, alongside logging and tracing.<br/>
+Metrics provide quantitative, numerical data about a system’s state and performance, typically sampled at regular intervals.
+They help track trends, resource usage, and system health over time.
+
+### Goals
+
+1. Compatibility with [OpenMetrics1.1](https://github.com/prometheus/OpenMetrics/blob/main/specification/OpenMetrics.md).
+2. Minimum dependencies.
+3. High performance with low memory footprint.
+4. Support for multiple exporters (e.g., Prometheus, JMX, CSV, etc.).
+5. Support client aggregations and resetting on export.
+6. Support plugging any external code into metrics.
+7. Provide SPI for easy configuration.
+
+### Architecture
+
+![metrics_architecture.jpeg](img/metrics_architecture.jpeg)
+
+### Key Concepts
+
+In many standards like **OpenMetrics** and **OpenTelemetry** metric has these key attributes:
+- Metadata (name, description, unit, type)
+- Labels (key-value string pairs)
+- Value(s) (numeric value)
```

Comment:
I believe this doc would benefit from a metric example. Alternatively, it can be in the metrics_details.md doc